feat(site): add bulk delete for tasks

mafredri opened the pull request:
This change implements bulk delete for tasks, closely copying UI and components from workspaces batch actions.

Since workspace batch actions are behind a feature, we add one for tasks as well.

Fixes coder/internal#1088

---

<img width="1185" height="900" alt="capture-eed2a76d" src="https://github.com/user-attachments/assets/f5352b9d-9504-4957-803b-dce7b530cbf8" />
<img width="1185" height="900" alt="capture-e8273b8c" src="https://github.com/user-attachments/assets/fa77535e-d1b0-4147-8718-f2a6c82ed523" />
<img width="1185" height="900" alt="capture-8d0ccc0c" src="https://github.com/user-attachments/assets/2287b1f3-c965-4f64-bc4a-7532a8ad419c" />

---

🤖 PR was written by Claude Sonnet 4.5 Thinking and reviewed by a human 🏂🏻

mafredri committed:
feat(site): add bulk delete for tasks

Fixes coder/internal#1088

mafredri committed:
add feature flag

DanielleMaywood commented on site/src/pages/TasksPage/TasksPage.tsx:
```diff
@@ -50,6 +63,53 @@ const TasksPage: FC = () => {
+	// biome-ignore lint/correctness/useExhaustiveDependencies: Reset on tab/filter changes.
+	useEffect(() => {
+		setCheckedTaskIds(new Set());
+	}, [tab.value, ownerFilter.value]);
```
I'll defer to Kayla but I'm not a huge fan of this. I think we should reset the checked task ids on tab/filter changes directly, instead of using `useEffect` for it.

aslilac replied:
yeah, this linter rule is yelling at you for a reason.

basically, this is a sign of bad component boundaries. if we only need this state _sometimes_ while the `TasksPage` is mounted, then we should just have a subcomponent which manages that state and which is only rendered when the state is needed and the related UI is actually on screen.

mafredri replied:
Fixed in [`c4827db` (#20905)](https://github.com/coder/coder/pull/20905/commits/c4827db3cc755a8aa38f4dcba254b63cc114cdbc)

aslilac reviewed (commented):
I feel like the text/stages could use a little more tweaking. I expect people to care a lot less about what terraform resources are being destroyed when deleting a task than they would when deleting workspaces that people actually ssh into.

and as danielle was saying, I definitely think we should convert the styles to tailwind. the goal has been "no new emotion styles, full stop" for quite a while. this would even be a good opportunity to convert the styles of the components these are taking from. leave the campsite better than you found it!

mafredri committed:
fix: change emotion to Tailwind

mafredri committed:
last created -> last updated

mafredri committed:
fix button

mafredri commented:
@aslilac @DanielleMaywood I believe I've addressed your feedback. PTAL (I haven't approved Chromatic changes yet so you can check those out too.)

Also: https://github.com/coder/coder/pull/20946

mafredri committed:
update testdata

mafredri committed:
add table batch actions test

mafredri committed:
remove row highlight on select

DanielleMaywood reviewed (approved):
From my side I'm happy

mafredri commented on site/src/pages/TasksPage/BatchDeleteConfirmation.stories.tsx:
```diff
@@ -0,0 +1,84 @@
+				name: "task-test-456",
+				display_name: "Add API Tests",
+				initial_prompt: "Add comprehensive tests for the API endpoints",
+				owner_name: "bob",
```
Bulk delete can be used by admins to delete other users tasks, but it may be true that we only allow displaying a single users task at the moment. I see no harm in keeping this for alignment with workspaces though.

mafredri committed:
pr feedback

mafredri committed:
pr feedback

mafredri committed:
remove useEffect per feedback

mafredri committed:
Merge branch 'main' into bulk-delete-feature-plan-
